src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -133,13 +136,35 @@ void OTARequestor::OnQueryImageResponse(void * context, const QueryImageResponse
             return;
         }
 
-        MutableByteSpan updateToken(requestorCore->mUpdateTokenBuffer);
-        CopySpanToMutableSpan(update.updateToken, updateToken);
-        requestorCore->mUpdateVersion = update.softwareVersion;
-        requestorCore->mUpdateToken   = updateToken;
+        if (response.softwareVersion.HasValue())
+        {
+
+            if (update.softwareVersion > currentSoftwareVersion)
+            {
+                ChipLogProgress(SoftwareUpdate, "Updating To Newer Version...");
+                MutableByteSpan updateToken(requestorCore->mUpdateTokenBuffer);
+                CopySpanToMutableSpan(update.updateToken, updateToken);
+                requestorCore->mUpdateVersion = update.softwareVersion;
+                requestorCore->mUpdateToken   = updateToken;
+
+                requestorCore->mOtaRequestorDriver->UpdateAvailable(
+                    update, System::Clock::Seconds32(response.delayedActionTime.ValueOr(0)));
+            }
+            else
+            {
+                ChipLogDetail(SoftwareUpdate, "Already UpToDate!!");
```

Comment:
```suggestion
                ChipLogDetail(SoftwareUpdate, "Already up to date");
```
Let's not use double exclamation marks in logs ;)